compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -9848,7 +9848,9 @@ private void handleSafeNavigation(BLangBlockStmt blockStmt, BLangAccessExpressio
             if (this.safeNavigationAssignment != null) {
                 this.safeNavigationAssignment.expr = addConversionExprIfRequired(accessExpr, tempResultVar.getBType());
             }
-            return;
+            if (Types.getReferredType(originalType).tag != TypeTags.JSON) {
```

Comment:
What's the plan for https://github.com/ballerina-platform/ballerina-lang/pull/39461/files#r1096973110?

Reply:
Can we use [isLaxFieldAccessAllowed](https://github.com/ballerina-platform/ballerina-lang/pull/39600/files#diff-51851f1451506fedd7c81b67a92282fada4eb00fe4728c70445d7262aed401f3R275) for this?

Reply:
@MaryamZi

Reply:
We need to try and see. What I meant was that we need to revisit this entire block and do the correct thing properly.